inspire_json_merger/postprocess.py

```diff
@@ -91,23 +92,81 @@ def postprocess_conflicts(conflicts, merged):
     """
     new_conflicts = []
     possible_duplicates = set()
-    for conflict in conflicts:
+
+    def _key_extractor(item):
+        """Extracts key used for sorting
+
+        Depending on type of conflict it extracts key increased by with artificial value depending in which group this conflict should appear
+            standard conflicts which have their own position are put in front
+            conflicts without position but with ORDER_KEY are put in the middle
+            conflicts without position and withouit ORDER_KEY are put at the end
+        """
+        _middle = 100000
+        _end = 1000000000000
+        if len(item[1]) > 1 and isinstance(item[1][1], int):
+            return item[1][1]
+        else:
+            if isinstance(item[2], dict) and ORDER_KEY in item[2]:
+                return item[2][ORDER_KEY] + _middle
+            else:
+                return _end
+
+    conflicts.sort(key=_key_extractor)
+    while conflicts:
+        conflict = conflicts.pop()
         conflict_type, conflict_location, conflict_content = conflict
         if conflict_type == "MANUAL_MERGE" and conflict_location[0] == "authors":
             new_conflict, merged, head = _process_author_manual_merge_conflict(conflict, merged)
             if new_conflict:
-                new_conflicts.append(new_conflict)
+                new_conflicts, conflicts = add_conflict(new_conflict, new_conflicts, conflicts)
                 possible_duplicates.add(head)
         elif not _is_conflict_duplicated(conflict, possible_duplicates):
             conflict_type, conflict_location, conflict_content = conflict
             if conflict_type == "ADD_BACK_TO_HEAD":
                 new_conflict, merged = _process_add_back_to_head(conflict, merged)
-                new_conflicts.append(new_conflict)
+                new_conflicts, conflicts = add_conflict(new_conflict, new_conflicts, conflicts)
             else:
                 new_conflicts.append(conflict)
     return new_conflicts, merged
 
 
+def add_conflict(conflict, processed_conflicts, unprocessed_conflicts):
+    """Adds conflict which added something to `merged` dict so positions for other conflicts should be updated
+
+    Args:
+        conflict(Conflict): curently added conflict
+        conflict_list(list): List of conflicts already added
+        unprocessed_conflicts(list): List of conflicts which will be added
+    """
+
+    processed_conflicts = update_conflicts_list(conflict, processed_conflicts)
+    unprocessed_conflicts = update_conflicts_list(conflict, unprocessed_conflicts)
```

Comment:
if you're updating both lists every time, why do you need to sort in the first place? because you're not updating the `ORDER_KEY`?

Reply:
We want to sort them at first as normal conflicts which do not require anything to be inserted into `merged` should go first, thanks to this latter it's easier to insert MANUAL_MERGES, also sorting them allows us to limit for how much records we re-calculate positions